tests/test_screens.py

```diff
@@ -537,3 +537,52 @@ def get_default_screen(self) -> Screen:
         assert len(app.screen_stack) == 1
         assert isinstance(app.screen_stack[0], CustomScreen)
         assert app.screen is app.screen_stack[0]
+
+
+async def test_worker_cancellation():
```

Comment:
I'm not really following this test or how it's related to this issue to be honest 😅 It doesn't really describe the issue and how it's testing it/ensuring it's fixed. Could you add comments to it to show where the issue occurred?